index.html

```diff
@@ -1089,12 +1094,8 @@ <h3>Processing Model</h3>
 "!PERFORMANCE-TIMELINE-2/#dfn-performance-entry-buffer">performance
 entry buffer</a>.</li>
 </ol>
-<p class="issue" data-number="82">This specification does not
-specify whether steps 20 and 21 should run before or after the
-<code>load</code> event of the resource—see <a href=
-"https://github.com/w3c/resource-timing/issues/82">issue 82</a> for
-related discussion.</p>
-<div class="issue" data-number="82"></div>
+<aside class="note">Steps 19 and 20 must be run before the <code>
```

Comment:
This is contrary to the conclusion in https://github.com/w3c/resource-timing/issues/82#issuecomment-340076036. 

Let's resolve https://github.com/w3c/resource-timing/issues/141#issuecomment-364099787 before we land this.

Reply:
Steps 19 and 20 do not involve much work at all and there is no reason for them to not run synchonously. I think what we wanted to delay in Chrome was executing the task in step 6 of https://www.w3.org/TR/performance-timeline-2/#dfn-queue-a-performanceentry
This is actually already spec'd as a low priority task, so Chrome is just trying to actually make it low priority by only running it when idle.

Reply:
We did not want to encourage use and assumption of entry being available before onload: if the PO entry is delivered asynchronously, we should apply same logic to queuing them into the buffer. Further, if we ever wanted to include processing time of the resource (e.g. https://github.com/w3c/resource-timing/issues/133), locking RT to be queued before onload event would then clash down the road.

Reply:
I don't see a problem with having a discrepancy between PO callbacks and entries in buffer - the fact that the callback is async is needed for performance reasons, right? And processing time sounds interesting, but I'm not sure how safe it is to delay pushing the entry by a lot, given that the existing implementations seem to add entries to the buffer ASAP. I'm working on tests to see how browsers behave with several types of resources, but if they all have entries by onload, then I would oppose to specing something that is contrary to that. This is a contentious point so I can remove this 'solution' to issue #141 for now but we should make a decision on this sometime.

Reply:
> I'm working on tests to see how browsers behave with several types of resources

Great. If we have this in time, let's review this on the next call. I'd like to get more eyeballs on this before we determine on how to proceed.

Reply:
Although we want this, we are changing the spec in a way that could break the web. I'd prefer to wait for the tests before we commit this text.

Reply:
Here are the tests https://github.com/w3c/web-platform-tests/pull/9459
They pass on Chrome https://chromium-review.googlesource.com/c/chromium/src/+/911868
But the WPT bot seems to be a little broken right now, the failures shown are from script crashing. But at least it runs on Firefox-nightly successfully. So now we just need results from Edge and Safari.

Reply:
Now that the WPT bot has been fixed, https://github.com/w3c/web-platform-tests/pull/9459 shows that ResourceTiming is exposed by the onload of a DOM object on all browsers.

Reply:
Since it seems this change is web compatible, I'm supportive of it, as it would increase predictability and simplify ResourceTiming based testing, if nothing else.

Reply:
> Although we want this, we are changing the spec in a way that could break the web. I'd prefer to wait for the tests before we commit this text.

@toddreifsteck to clarify, I don't believe we're breaking anything here.. Existing spec language says the behavior is *not* defined and entries may be added during or after onload. The proposed language is changing that to lock in "before onload". 

> Since it seems this change is web compatible, I'm supportive of it, as it would increase predictability and simplify ResourceTiming based testing, if nothing else.

That's a good argument for adopting this. However, I think this might also make https://github.com/w3c/resource-timing/issues/133 tricky. As of today, it's easy and simple for us to add another timestamp that captures duration of processing / onload time, but if we lock-in the queue step to prior to onload then, well.. We'll need some separate entry.

Reply:
> However, I think this might also make #133 tricky

Good point! At the same time, if as a result of https://github.com/w3c/resource-timing/issues/137 we'll end up making the entries mutable (similar to PerformanceNavigationTiming, as discussed on the call) that'd be less of an issue.

Reply:
True, but that's also rather odd behavior. In this case we're talking about entries available via global timeline, not PerfObserver. You could make the case that we already surface "incomplete" entries via NavTiming, but this just perpetuates the confusion.

Reply:
From an analytics provider's perspective, incomplete entries (marked as such, with an event indicating mutation or completion) are better than no entries.

Regardless, this seems like something that all current implementations agree on, so changing this behavior in the future wouldn't be harder than it is right now. (and I suspect it's already not web compatible to change it, but could be proven wrong)

Reply:
> Regardless, this seems like something that all current implementations agree on, so changing this behavior in the future wouldn't be harder than it is right now.

Fair enough, I think you have me convinced. Let's get a final ack from wider group on our next call, and we can proceed?

Reply:
SGTM